lib/rules/no-deprecated-router-transition-methods.js

```diff
@@ -0,0 +1,193 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+const { getImportIdentifier } = require('../utils/import');
+const Stack = require('../utils/stack');
+const types = require('../utils/types');
+const decoratorUtils = require('../utils/decorators');
+
+function getBaseFixSteps(fixer, context, currentClass) {
+  const fixSteps = [];
+  const sourceCode = context.getSourceCode();
+  let serviceInjectImportName = currentClass.serviceInjectImportName;
+  let routerServicePropertyName = currentClass.routerServicePropertyName;
+
+  if (!serviceInjectImportName) {
+    fixSteps.push(
+      fixer.insertTextBefore(
+        sourceCode.ast,
+        "import { inject as service } from '@ember/service';\n"
+      )
+    );
+
+    serviceInjectImportName = 'service';
+  }
+
+  if (!routerServicePropertyName) {
+    fixSteps.push(
+      fixer.insertTextBefore(
+        currentClass.node.body.body[0],
+        `@${serviceInjectImportName}('router') router;\n`
+      )
+    );
+
+    routerServicePropertyName = 'router';
+  }
+
+  return { fixSteps, routerServicePropertyName };
+}
+
+const createErrorMessage = (methodUsed, desiredMethod, moduleType) =>
+  `Calling "${methodUsed}" in ${moduleType}s is deprecated, call ${desiredMethod} on the injected router service instead.`;
```

Comment:
Instead of this, you can try using `messageId` instead (https://eslint.org/docs/latest/developer-guide/working-with-rules#messageids) and then you can pass the necessary dynamic data

Reply:
Cool, I'm not too familiar with custom messages, this will be helpful and I aim to fix this today.